dockers/docker-ptf/gnxi-patches/0007-Fix-py_gnmicli.py-POLL-mode.patch

```diff
@@ -0,0 +1,91 @@
+From d35bc8b5b64361d981c902b82f097de0d44cc2b1 Mon Sep 17 00:00:00 2001
+From: Zain Budhwani <zainbudhwani@microsoft.com>
+Date: Mon, 21 Apr 2025 20:55:38 +0000
+Subject: [PATCH] Fix py_gnmicli.py POLL mode
+
+---
+ gnmi_cli_py/py_gnmicli.py | 17 +++++++++++++++++
+ 1 file changed, 17 insertions(+)
+
+diff --git a/gnmi_cli_py/py_gnmicli.py b/gnmi_cli_py/py_gnmicli.py
+index 0ea6f3d..ba3308b 100644
+--- a/gnmi_cli_py/py_gnmicli.py
++++ b/gnmi_cli_py/py_gnmicli.py
+@@ -39,6 +39,7 @@ import ssl
+ import sys
+ import string
+ import six
++import time
+ import datetime
+ try:
+   import gnmi_pb2
+@@ -47,6 +48,7 @@ except ImportError:
+         'eg, pip install -r requirements.txt')
+ import gnmi_pb2_grpc
+ import grpc
++from gnmi_pb2 import Poll
+ 
+ __version__ = '0.5'
+ 
+@@ -153,6 +155,8 @@ def _create_parser():
+                       required=False, action='store_true')
+   parser.add_argument('--interval', default=10000, type=int,
+                       help='sample interval in millisecond (default: 10000ms)')
++  parser.add_argument('--polling_interval', default=10, type=int,
++                      help='polling interval in second (default: 10s)')
+   parser.add_argument('--timeout', type=int, help='subscription'
+                       'duration in seconds (default: none)')
+   parser.add_argument('--heartbeat', default=0, type=int, help='heartbeat interval (default: None)')
+@@ -161,6 +165,7 @@ def _create_parser():
+   parser.add_argument('--submode', default=2, type=int,
+                       help='subscription mode [0=TARGET_DEFINED, 1=ON_CHANGE, 2=SAMPLE]')
+   parser.add_argument('--update_count', default=0, type=int, help='Max number of streaming updates to receive. 0 means no limit.')
++  parser.add_argument('--sync_count', default=0, type=int, help='Max number of sync responses to receive. 0 means no limit.')
```

Comment:
>0 [](http://example.com/codeflow?start=107&length=1)

"-1 means no limit" make better sense, and leading to simplified implementation. Could you explore?

Reply:
Changed it to -1 no limit instead of 0